### ISSUE TITLE
DB/Item: Carrot on a Stick/Riding Crop

### ISSUE THREAD

**Issue by Staleness89** [TrinityCore/TrinityCore#18102]
**Description**:
Carrot on a Stick, Riding Crop and Skybreaker's Whip all should give mount speed bonuses only to a certain level. Right now, they have no bonuses.

http://www.wowhead.com/item=11122/carrot-on-a-stick
http://www.wowhead.com/item=25653/riding-crop
http://www.wowhead.com/item=32863/skybreaker-whip

**Current behaviour**: Tell us what happens
Nothing changes when item is equipped.

**Expected behaviour**: Tell us what should happen instead
You should receive a boost to mount speed whenever these items are equipped.

**Steps to reproduce the problem**:

1.Equip one of the 3 items listed above. 
2. Run around.
3. Notice lack of mount speed bonus (hard to tell with the carrot lol)

**Branch(es)**: 3.3.5 / 6.x (Select the branch(es) affected by this issue) 
Both
**TC hash/commit**:  
b00fa41
**TDB version**:  
TDB 335.62
**Operating system**:  
Win 10

**SUGGESTED FIX**
```SQL
--
DELETE FROM `spell_linked_spell` WHERE `spell_trigger` = 48777 AND `spell_effect` = 48402;
INSERT INTO `spell_linked_spell` (`spell_trigger`, `spell_effect`, `type`, `comment`) VALUES (48777, 48402, 2, 'Carrot on a Stick');

DELETE FROM `spell_linked_spell` WHERE `spell_trigger`= 48776 AND `spell_effect` = 48383;
INSERT INTO `spell_linked_spell` (`spell_trigger`, `spell_effect`, `type`, `comment`) VALUES (48776, 48383, 2, 'Riding Crop/Skybreaker Whip');
```
However this fix is missing the level check for both spells. Was hoping someone could come up with something in the DB conditions table for them.

**Comment by Killyana** [comment]
I tried to display an error message via `ErrorTextId` and `ErrorType` but it doesn't work
Condition: 
``` sql
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=17 AND `SourceEntry` IN (48777, 48776);
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorTextId`, `ScriptName`, `Comment`, `ErrorType`) VALUES
(17,0,48777,0,0, 27,0,71,2,0,0,86,'','Mount Speed',36),
(17,0,48776,0,0, 27,0,71,2,0,0,86,'','Mount Speed',36);

**Comment by Shauren** [comment]
Don't do it like that - conditions are a bad way to handle it, they cause the spell cast to fail but the cast shouldn't happen in the first place (and no error should be displayed to player)

**Comment by Killyana** [comment]
In this case `ConditionTarget`=0 ( Caster of the spell ) so the spell is not casted, probably for this reason there's no error displayed.

**Comment by Shauren** [comment]
ConditionTarget has nothing to do with the spell being cast or not. The entire check is done inside `Spell::CheckCast` - at this point the cast is already initialized and will cause SMSG_CAST_FAILED to be sent. The reason you don't see any error ingame is because of a check in `Spell::SendCastResult` that silently converts all errors into SPELL_FAILED_DONT_REPORT if the cast has triggered flag TRIGGERED_DONT_REPORT_CAST_ERROR (by default on for all triggered spells)

**Comment by Staleness89** [comment]
yeah I wasn't sure if you could do it in conditions.  I would make the script but I'm not too familiar with all that yet.. This is better than nothing, though, at least for me.

**Comment by CraftedRO** [comment]
835e2918319e Those items doesn't seem to work at all :

https://user-images.githubusercontent.com/24683355/213568091-6d3afb5e-cd5e-4de9-8e87-9a2de76e3df6.mp4
